pkg/util/parse.go

```diff
@@ -113,3 +114,29 @@ func convert(original map[string]interface{}) (err error, typed map[string]map[s
 	}
 	return nil, m2
 }
+
+func appearsToReferenceVariableInAnotherYaml(s string) bool {
+	if containsColon(s) {
+		// A path to another yaml can never ever contain a colon. Therefore, bailing out if s contains one.
+		return false
+	}
+	if doesNotReferenceKnownVariable(s) {
+		// As of right now there's only a limited number of variables that can be referenced. If s points to something else let's bail out here.
+		return false
+	}
+	return true
+}
+
+func referencesJSON(s string) bool {
+	return strings.HasSuffix(s, ".json")
```

Comment:
I'm wondering if there is any usecase/type of Dynatrace config in which I'd pass a .json filename as actual content? 
For our JSON references, I think they will always be in the initial `config:` object. 

Not sure if we should be stricter in this check already, or just wait and see if it becomes an issue. 
If we wait and see, maybe placing this comment in the method might be helpful? 
@didiladi ?

Reply:
Hmm, that's a good point @UnseenWizzard :+1: I like the idea to check for the `config` object.

Reply:
But that could also be done in a follow-up PR in my opinon.

Reply:
I guess we should sanity check that the current handling at least doesn't cause in issue in any of our config repos?

If not I agree, we may move to base it on the config as a follow up only if needed

Reply:
I just found an example in our config repo: the extension name usually ends with `.json`. Ok, so we need to add the check in this PR. Thanks for being so persistent @UnseenWizzard :smile:

Reply:
And we don't run into this problem in the integration tests, because the name of our example doesn't end with `.json`: https://github.com/dynatrace-oss/dynatrace-monitoring-as-code/blob/main/cmd/monaco/test-resources/integration-all-configs/project/extension/plugins.yaml#L6

Reply:
We could solve this issue by requiring some sort of prefix for variable references.
Something like `$/other/file.id` / `$other/file.name`. 

And it would maybe make sense to only allow for `/` in config references. We
could then after detection of a reference call `filepath.FromSlash` to get the 
os specific file path.

Reply:
persisting a quick discussion with @patrickpichler : 

We think for now we should limit to replacing paths in config:, not on all .jsons

Long term we should discuss the suggestions on how we can better handle general variables/references to paths (e.g. marking them)

Reply:
I added a new discussion ticket for this #98

Reply:
@didiladi @UnseenWizzard @patrickpichler : I've changed it so that references to JSONs are only replaced if they appear in the YAML config section. Hope I understood it correctly?